backend/src/main/kotlin/database/PostgreRepository.kt

```diff
@@ -0,0 +1,164 @@
+package database
+
+import entities.Season
+import entities.SeasonHalf.Companion.toSeasonHalf
+import entities.Store
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.`java-time`.date
+import org.jetbrains.exposed.sql.transactions.TransactionManager
+import org.jetbrains.exposed.sql.transactions.transaction
+import java.sql.SQLException
+import java.time.Year
+
+class PostgreRepository(private val database: Database) : Repository {
+    private object StoreTable : Table("store") {
+        val id = long("id")
+        val code = varchar("code", 3000).nullable()
+        val description = varchar("description", 3000).nullable()
+        val name = varchar("name", 100)
+        val openingDate = date("openingDate").nullable()
+        val storeType = varchar("storeType", 100).nullable()
+        val specialField1 = varchar("specialField1", 1500).nullable()
+        val specialField2 = varchar("specialField2", 1500).nullable()
+
+        override val primaryKey = PrimaryKey(id)
+    }
+
+    private object SeasonTable : Table("season") {
+        val half = varchar("half", 2)
+        val year = integer("year")
+        override val primaryKey = PrimaryKey(half, year)
+    }
+
+    private object StoreSeasonTable : Table("stores_seasons") {
+        val storeId = long("storeId").references(StoreTable.id, onDelete = ReferenceOption.CASCADE)
+        val half = varchar("half", 2)
```

Comment:
should this be a reference to the `SeasonTable` instead?

Reply:
It should. Although, I found no satisfactory way to do a composite foreign key using Jetbrains Exposed ORM. 
The only way to do this was manually (take a look at [this open issue](https://github.com/JetBrains/Exposed/issues/511))


This is related to the sketchy FK manipulation that is happening at the end of the file

Reply:
@alramalho there are some good reasons why frameworks like Exposed and Hibernate do not really support composite keys. 

What do you think about the pros and cons of following alternative table:

```SQL
CREATE TABLE store_season_rel (
    id               BIGSERIAL NOT NULL,
    store_id     BIGINT        NOT NULL REFERENCES stores (id),
    season_id BIGINT        NOT NULL REFERENCES seasons (id),
    CONSTRAINT uc_relation UNIQUE (store_id, season_id)
);
```

Reply:
That is a nice way to rewrite the table, working around the need of a composite alternate key. With hindsight, I would go with that path, instead of manually inserting this logic through SQL commands, as I did.